src/components/forms/PractitionerForm/UserIdSelect/index.tsx

```diff
@@ -132,6 +132,37 @@ export const UserIdSelect = (props: Props) => {
     return allUsers;
   };
 
+  /** pull all practioners */
+  const getAllPractitioners = async () => {
+    let filterParams = {
+      pageNumber: 1,
+      pageSize: PRACTITIONER_REQUEST_PAGE_SIZE,
+    };
+    const allPractitioners = [];
+    let response: Practitioner[];
+    let serverError = false;
+    const service = new props.serviceClass(OPENSRP_PRACTITIONER_ENDPOINT);
+
+    do {
```

Comment:
I wonder if there is a way to do this with chained promises to avoid awaiting?  Some recursion would need to be involved, I think.  Do you think it is worth it to pursue that?

Reply:
I think it can be possible if we had the endpoint for returning total counts ready. Currently i have no way of knowing when to stop unless i wait for the results.

Reply:
I think its probably possible to construct the promise chain and in the `then` block of each promise check if there are any records returned.  You'd then stop chaining when there are no records returned.

The advantage here is that everything is async and we have the store getting populated as soon as the first promise resolves.